### ISSUE TITLE
Autoptimize raises exception in a "string is valid regex" check

### ISSUE THREAD

**Issue by diddledan** [iolevel/wpdotnet-sdk#26]
**Describe the bug**
After compiling autoptimize and enabling in my WPdotNet site's dashboard, I receive an exception from autoptimize-specific code which is designed to test whether a string is a valid regular expression.

The function from `autoptimizeUtils.php` is below:

```php
    /**
     * Returns true if the string is a valid regex.
     *
     * @param string $string String, duh.
     *
     * @return bool
     */
    public static function str_is_valid_regex( $string )
    {
        set_error_handler( function() {}, E_WARNING );
        $is_regex = ( false !== preg_match( $string, '' ) );
        restore_error_handler();

        return $is_regex;
    }
```

My instance is calling that function with `$string` set to `<!--[if` and the exception states `No ending delimiter '[' found`.

**To Reproduce**
A reduced testcase is the following PHP file which can be added to an empty Peachpie Console application project:

```php
<?php

function str_is_valid_regex( $string )
{
    set_error_handler( function() {}, E_WARNING );
    $is_regex = ( false !== preg_match( $string, '' ) );
    restore_error_handler();

    return $is_regex;
}

print_r(str_is_valid_regex('<!--[if'));
```

The test case is also on [GitHub at diddledan/Peachpie-AutoptimizeTestCase](https://github.com/diddledan/Peachpie-AutoptimizeExceptionTestcase)

**Expected behavior**
Autoptimize expects that the function return false, without breaking execution with a thrown exception, for the input `<!--[if`.

**Screenshots**

**System information:**
 - OS: Windows 10
 - Version [e.g. 1.04]: Peachpie.NET.Sdk/0.9.36

**Stack trace:**
The stacktrace here is from my reduced testcase:

```plain
   at Pchp.Library.PerlRegex.RegexParser.TrimDelimiters(String re, Int32 end)
   at Pchp.Library.PerlRegex.RegexParser.Parse(String re, RegexOptions op)
   at Pchp.Library.PerlRegex.Regex..ctor(String pattern, RegexOptions options, TimeSpan matchTimeout, Boolean useCache)
   at Pchp.Library.PerlRegex.Regex..ctor(String pattern)
   at Pchp.Library.PCRE.preg_match(Context ctx, String pattern, String subject)
   at <Root>.program_php.str_is_valid_regex(Context <ctx>, PhpValue string) in C:\Users\yabea\source\repos\AutoptimizeExceptionTestcase\AutoptimizeExceptionTestcase\program.php:line 6
   at <Root>.program_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in C:\Users\yabea\source\repos\AutoptimizeExceptionTestcase\AutoptimizeExceptionTestcase\program.php:line 12
   at <Script>.Main(String[] args)
```

**Comment by bfistein** [comment]
@diddledan would you mind having a look if the latest release of PeachPie didn't fix this? We implemented a complete overhaul of PCRE. Thank you!

**Comment by jakubmisek** [comment]
thanks, has been fixed with latest peahchpie > 0.9.600
